pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/NullableSingleInputAggregationFunction.java

```diff
@@ -0,0 +1,368 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.aggregation.function;
+
+import it.unimi.dsi.fastutil.floats.FloatConsumer;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Consumer;
+import java.util.function.DoubleConsumer;
+import java.util.function.Function;
+import java.util.function.LongConsumer;
+import javax.annotation.Nullable;
+import org.apache.pinot.common.request.context.ExpressionContext;
+import org.apache.pinot.core.common.BlockValSet;
+import org.roaringbitmap.IntConsumer;
+import org.roaringbitmap.IntIterator;
+import org.roaringbitmap.RoaringBitmap;
+
+
+public abstract class NullableSingleInputAggregationFunction<I, F extends Comparable>
+    extends BaseSingleInputAggregationFunction<I, F> {
+  protected final boolean _nullHandlingEnabled;
+
+  public NullableSingleInputAggregationFunction(ExpressionContext expression, boolean nullHandlingEnabled) {
+    super(expression);
+    _nullHandlingEnabled = nullHandlingEnabled;
+  }
+
+
+  @FunctionalInterface
+  public interface BatchConsumer {
```

Comment:
I can see them simplify the code, but from past experience, the functional interface can significant impact the performance (at least for java 8). Aggregation function is one of the most performance sensitive code, and these function are applied on a per value basis. Can we do some micro benchmark to ensure there is no overhead?

Reply:
I've created a benchmark in another PR (marked as draft). See https://github.com/apache/pinot/pull/12354